// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package mobility.collision;

import "collision/grid_common.proto";

// Serializes mobility::collision::OccupancyGrid in a more efficient way.
// The occupancy is stored as compressed PNG.
// Next ID: 5
message OccupancyGridProto {
  // State of a grid cell.
  enum OccupancyStatus {
    UNOCCUPIED = 0;
    OCCUPIED = 1;
    UNKNOWN = 2;
  }

  // Occupancy payload.
  oneof occupancy_data {
    bytes occupancy_png = 3;             // The PNG compressed image.
    OccupancyStatus occupancy_fill = 4;  // Occupancy value for solid fill.
    bytes occupancy_raw = 5;             // The raw image.
  }

  GridFrameProto grid_frame = 1;
  GridRangeProto grid_range = 2;
}
